pkg/client/client.go

```diff
@@ -0,0 +1,150 @@
+// SPDX-FileCopyrightText: 2023, SAS Institute Inc., Cary, NC, USA.  All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/sassoftware/event-provenance-registry/pkg/storage"
+	"github.com/sassoftware/event-provenance-registry/pkg/utils"
+)
+
+var logger = utils.MustGetLogger("client", "client")
+
+// ensure it implements the interface
+var _ Contract = &Client{}
+
+// Contract handles communications with the EPR service
+type Contract interface {
+	CreateEvent(e *storage.Event) (string, error)
+	CreateEventReceiver(er *storage.EventReceiver) (string, error)
+	CreateEventReceiverGroup(erg *storage.EventReceiverGroup) (string, error)
+	ModifyEventReceiverGroup(erg *storage.EventReceiverGroup) (string, error)
+	QueryEventByID(id string, fields []string) (*storage.Event, error)
+	QueryEventReceiverByID(id string, fields []string) (*storage.EventReceiver, error)
+	QueryEventReceiverGroupByID(id string, fields []string) (*storage.EventReceiverGroup, error)
+	IDSearch(params map[string]interface{}) (string, error)
+	Search(queryName string, queryFor string, params map[string]interface{}, fields []string) (string, error)
+	SearchEvents(params map[string]interface{}, fields []string) (string, error)
+	SearchEventsObj(params map[string]interface{}, fields []string) ([]storage.Event, error)
```

Comment:
I'm wondering if it's worth having both of these? If you want only the ID, it's easy enough to request only that in graphql.